s3auth-relay/src/main/java/com/s3auth/relay/HttpThread.java

```diff
@@ -193,23 +195,24 @@ public long dispatch() throws InterruptedException {
                     }
                 }
             } else {
-                bytes = HttpThread.failure(
+                bytes = this.failure(
                     new HttpException(
                         HttpURLConnection.HTTP_BAD_METHOD,
                         "only GET and HEAD methods are supported at the moment"
                     ),
                     socket
                 );
-                Logger.info(this, "#run(): failure sent to %s", socket);
             }
         } catch (final HttpException ex) {
-            bytes = HttpThread.failure(ex, socket);
+            Logger.warn(this, "#run(): %s", ExceptionUtils.getStackTrace(ex));
```

Comment:
you can use a decorator of `Logger`, see http://log.jcabi.com/decors.html. in this case: `Logger.warn(this, "#run(): %[exception]s")`